cellprofiler/modules/noisereduction.py

```diff
@@ -0,0 +1,107 @@
+"""
+
+Noise reduction
+
+"""
+
+import cellprofiler.image
+import cellprofiler.module
+import cellprofiler.setting
+import skimage.restoration
+import skimage.util
+
+
+class NoiseReduction(cellprofiler.module.Module):
```

Comment:
Should we add support for [total variation denoising](https://en.wikipedia.org/wiki/Total_variation_denoising)?

It’s not non-local, but it’s faster.